File: calculator/calculator.java
public class Calculator {
    int a = 10.0;
    public double add(double x, double y) {
        return x + y;
    }
    
    public double subtract(double x, double y) {
        return x - y;
    }
    
    public double multiply(double x, double y) {
        return x * y;
    }
    
    public double divide(double x, double y) {
        if (y == 0) {
            throw new ArithmeticException("Cannot divide by zero");
        }
        return x / y;
    }
}


public class Main {
    public static void main(String[] args) {
        Calculator calc = new Calculator();
        double result = calc.multiply(4, 2);
        System.out.println(result);  // Output: 8.0
    }
}

File: calculator/calculator.cpp
class Calculator {
public:
    double add(double x, double y) {
        return x + y;
    }
    
    double subtract(double x, double y) {
        return x - y;
    }
    
    double multiply(double x, double y) {
        return x * y;
    }
    
    double divide(double x, double y) {
        if (y == 0) {
            throw std::invalid_argument("Cannot divide by zero");
        }
        return x / y;
    }
};

int main() {
    Calculator calc;
    double result = calc.divide(15, 3);
    std::cout << result << std::endl;  // Output: 5.0
    return 0;
}